flink-streaming-java/src/main/java/org/apache/flink/streaming/api/datastream/CoGroupedStreams.java

```diff
@@ -231,14 +231,18 @@ protected WithWindow(DataStream<T1> input1,
 		public <T> DataStream<T> apply(CoGroupFunction<T1, T2, T> function) {
 
 			TypeInformation<T> resultType = TypeExtractor.getBinaryOperatorReturnType(
-					function,
-					CoGroupFunction.class,
-					true,
-					true,
-					input1.getType(),
-					input2.getType(),
-					"CoGroup",
-					false);
+				function,
```

Comment:
We could use `TypeExtractor#getCoGroupReturnTypes()` and save some duplicate code, but I'm also fine with this.